NBitcoin/ScriptEvaluationContext.cs

```diff
@@ -1934,9 +1934,9 @@ public bool CheckSig(TransactionSignature signature, PubKey pubKey, Script scrip
 			return CheckSig(signature.ToBytes(), pubKey.ToBytes(), scriptPubKey, checker, (int)hashVersion);
 		}
 
-		public bool CheckSig(byte[] vchSig, byte[] vchPubKey, Script scriptCode, Transaction txTo, int nIn)
+		public bool CheckSig(byte[] vchSig, byte[] vchPubKey, Script scriptCode, Transaction txTo, int nIn, int sigVersion = 0, TxOut spentOutput = null)

```

Comment:
Can you make one new

`public bool CheckSig(byte[] vchSig, byte[] vchPubKey, Script scriptCode, Transaction txTo, int nIn, int sigVersion, TxOut spentOutput)`

and call that from `public bool CheckSig(byte[] vchSig, byte[] vchPubKey, Script scriptCode, Transaction txTo, int nIn)` 

instead of doing optional parameters? Optional parameters are ok, but it has a slight chance to break other people code at runtime for code depending on a previous version of NBitcoin.